include/oneapi/dpl/pstl/onedpl_config.h

```diff
@@ -333,4 +340,47 @@
 #    define _ONEDPL_STD_RANGES_ALGO_CPP_FUN 0
 #endif
 
+// -- Configure heterogeneous backends --
+
+#if defined(ONEDPL_FPGA_DEVICE)
+#    undef _ONEDPL_FPGA_DEVICE
+#    define _ONEDPL_FPGA_DEVICE ONEDPL_FPGA_DEVICE
+#endif
+
+#if !defined(ONEDPL_ALLOW_DEFERRED_WAITING)
+#    define ONEDPL_ALLOW_DEFERRED_WAITING 0
+#endif
+
+#if defined(ONEDPL_FPGA_EMULATOR)
+#    undef _ONEDPL_FPGA_EMU
+#    define _ONEDPL_FPGA_EMU ONEDPL_FPGA_EMULATOR
+#endif
+
+#if defined(ONEDPL_USE_PREDEFINED_POLICIES)
+#    undef _ONEDPL_PREDEFINED_POLICIES
+#    define _ONEDPL_PREDEFINED_POLICIES ONEDPL_USE_PREDEFINED_POLICIES
+#elif !defined(_ONEDPL_PREDEFINED_POLICIES)
+#    define _ONEDPL_PREDEFINED_POLICIES 1
+#endif
+
+#if _ONEDPL_BACKEND_SYCL
+// Include sycl specific options
+// FPGA doesn't support sub-groups
+#    if !(_ONEDPL_FPGA_DEVICE)
+#        define _USE_SUB_GROUPS 1
+#        define _USE_GROUP_ALGOS 1
+#    endif
+
+#    define _USE_RADIX_SORT (_USE_SUB_GROUPS && _USE_GROUP_ALGOS)
+
+// Compilation of a kernel is requiried to obtain valid work_group_size
+// when target devices are CPU or FPGA emulator. Since CPU and GPU devices
+// cannot be distinguished during compilation, the macro is enabled by default.
+#    if !defined(_ONEDPL_COMPILE_KERNEL)
+#        define _ONEDPL_COMPILE_KERNEL 1
+#    endif
```

Comment:
It would be good to mention in the comment when/where the macro can be turned off.

Reply:
I think that it deserves a separate investigation, because I do not know the exact root cause why `_ONEDPL_COMPILE_KERNEL` can be turned off, which is necessary to answer that question.

What I know is that we compile a kernel to reliably understand the maximum work-group size according to the resource allocation per work-item, and for some reason (it looks like due to empirical findings) it can be avoidable for GPUs.

Would you mind if a create an issue to address that problem?

Reply:
I think that it deserves a separate investigation, because I have no answer and it is not related directly to the PR.

What I know is:
 - We compile a kernel to reliably understand the maximum work-group size. This kernel-specific size may be smaller than the maximum supported work-group size by a device, e.g. due to using a large amount of registers or SLM.
 - For GPUs, it seems, we have found some empirical rules to avoiding out-of-resources issues. I do not know if they are applicable, e.g. to CPU, FPGA, or other GPU kernels. Even though, I do not expect that everything will work out-of-the-box for GPU if the macro is set to 0.
 - Explicit kernel compilation introduces overhead, impactful on small problem sizes. I do not know which sizes exactly, though.
 - Some implementations may not provide kernels (e.g. AdaptiveCPP), so it should be turned off there.

Would you mind if I create a separate issue to improve the comment?

Reply:
I have traced it back to the very first snapshot of oneDPL source files, and even there the code was like today. I can guess that it was added as an internal knob to manually disable kernel compilation.

Yes, let's revisit it later, and not for improving the comment but for improving the definition or removing it altogether.